// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: tzero/v1/payment_intent/recipient/recipient.proto

package recipient

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"

	common "github.com/t-0-network/provider-sdk-go/api/gen/proto/tzero/v1/common"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort

	_ = common.PaymentMethodType(0)
)

// Validate checks the field values on CreatePaymentIntentRequest with the
// rules defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *CreatePaymentIntentRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on CreatePaymentIntentRequest with the
// rules defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// CreatePaymentIntentRequestMultiError, or nil if none found.
func (m *CreatePaymentIntentRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *CreatePaymentIntentRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for PaymentReference

	// no validation rules for PayInCurrency

	if all {
		switch v := interface{}(m.GetPayInAmount()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, CreatePaymentIntentRequestValidationError{
					field:  "PayInAmount",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, CreatePaymentIntentRequestValidationError{
					field:  "PayInAmount",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetPayInAmount()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return CreatePaymentIntentRequestValidationError{
				field:  "PayInAmount",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	// no validation rules for PayOutCurrency

	if all {
		switch v := interface{}(m.GetPayOutMethod()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, CreatePaymentIntentRequestValidationError{
					field:  "PayOutMethod",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, CreatePaymentIntentRequestValidationError{
					field:  "PayOutMethod",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetPayOutMethod()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return CreatePaymentIntentRequestValidationError{
				field:  "PayOutMethod",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	if len(errors) > 0 {
		return CreatePaymentIntentRequestMultiError(errors)
	}

	return nil
}

// CreatePaymentIntentRequestMultiError is an error wrapping multiple
// validation errors returned by CreatePaymentIntentRequest.ValidateAll() if
// the designated constraints aren't met.
type CreatePaymentIntentRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m CreatePaymentIntentRequestMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m CreatePaymentIntentRequestMultiError) AllErrors() []error { return m }

// CreatePaymentIntentRequestValidationError is the validation error returned
// by CreatePaymentIntentRequest.Validate if the designated constraints aren't met.
type CreatePaymentIntentRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e CreatePaymentIntentRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e CreatePaymentIntentRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e CreatePaymentIntentRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e CreatePaymentIntentRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e CreatePaymentIntentRequestValidationError) ErrorName() string {
	return "CreatePaymentIntentRequestValidationError"
}

// Error satisfies the builtin error interface
func (e CreatePaymentIntentRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sCreatePaymentIntentRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = CreatePaymentIntentRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = CreatePaymentIntentRequestValidationError{}

// Validate checks the field values on CreatePaymentIntentResponse with the
// rules defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *CreatePaymentIntentResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on CreatePaymentIntentResponse with the
// rules defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// CreatePaymentIntentResponseMultiError, or nil if none found.
func (m *CreatePaymentIntentResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *CreatePaymentIntentResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for PaymentIntentId

	for idx, item := range m.GetPayInPaymentMethods() {
		_, _ = idx, item

		if all {
			switch v := interface{}(item).(type) {
			case interface{ ValidateAll() error }:
				if err := v.ValidateAll(); err != nil {
					errors = append(errors, CreatePaymentIntentResponseValidationError{
						field:  fmt.Sprintf("PayInPaymentMethods[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			case interface{ Validate() error }:
				if err := v.Validate(); err != nil {
					errors = append(errors, CreatePaymentIntentResponseValidationError{
						field:  fmt.Sprintf("PayInPaymentMethods[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			}
		} else if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return CreatePaymentIntentResponseValidationError{
					field:  fmt.Sprintf("PayInPaymentMethods[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	if len(errors) > 0 {
		return CreatePaymentIntentResponseMultiError(errors)
	}

	return nil
}

// CreatePaymentIntentResponseMultiError is an error wrapping multiple
// validation errors returned by CreatePaymentIntentResponse.ValidateAll() if
// the designated constraints aren't met.
type CreatePaymentIntentResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m CreatePaymentIntentResponseMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m CreatePaymentIntentResponseMultiError) AllErrors() []error { return m }

// CreatePaymentIntentResponseValidationError is the validation error returned
// by CreatePaymentIntentResponse.Validate if the designated constraints
// aren't met.
type CreatePaymentIntentResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e CreatePaymentIntentResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e CreatePaymentIntentResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e CreatePaymentIntentResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e CreatePaymentIntentResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e CreatePaymentIntentResponseValidationError) ErrorName() string {
	return "CreatePaymentIntentResponseValidationError"
}

// Error satisfies the builtin error interface
func (e CreatePaymentIntentResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sCreatePaymentIntentResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = CreatePaymentIntentResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = CreatePaymentIntentResponseValidationError{}

// Validate checks the field values on GetQuoteRequest with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *GetQuoteRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetQuoteRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetQuoteRequestMultiError, or nil if none found.
func (m *GetQuoteRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *GetQuoteRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for PayInCurrency

	if all {
		switch v := interface{}(m.GetPayInAmount()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, GetQuoteRequestValidationError{
					field:  "PayInAmount",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, GetQuoteRequestValidationError{
					field:  "PayInAmount",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetPayInAmount()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return GetQuoteRequestValidationError{
				field:  "PayInAmount",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	// no validation rules for PayOutCurrency

	// no validation rules for PayInPaymentMethod

	// no validation rules for PayOutPaymentMethod

	if len(errors) > 0 {
		return GetQuoteRequestMultiError(errors)
	}

	return nil
}

// GetQuoteRequestMultiError is an error wrapping multiple validation errors
// returned by GetQuoteRequest.ValidateAll() if the designated constraints
// aren't met.
type GetQuoteRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetQuoteRequestMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetQuoteRequestMultiError) AllErrors() []error { return m }

// GetQuoteRequestValidationError is the validation error returned by
// GetQuoteRequest.Validate if the designated constraints aren't met.
type GetQuoteRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetQuoteRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetQuoteRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetQuoteRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetQuoteRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetQuoteRequestValidationError) ErrorName() string { return "GetQuoteRequestValidationError" }

// Error satisfies the builtin error interface
func (e GetQuoteRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetQuoteRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetQuoteRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetQuoteRequestValidationError{}

// Validate checks the field values on GetQuoteResponse with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *GetQuoteResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetQuoteResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetQuoteResponseMultiError, or nil if none found.
func (m *GetQuoteResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *GetQuoteResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	switch v := m.Result.(type) {
	case *GetQuoteResponse_Quote_:
		if v == nil {
			err := GetQuoteResponseValidationError{
				field:  "Result",
				reason: "oneof value cannot be a typed-nil",
			}
			if !all {
				return err
			}
			errors = append(errors, err)
		}

		if all {
			switch v := interface{}(m.GetQuote()).(type) {
			case interface{ ValidateAll() error }:
				if err := v.ValidateAll(); err != nil {
					errors = append(errors, GetQuoteResponseValidationError{
						field:  "Quote",
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			case interface{ Validate() error }:
				if err := v.Validate(); err != nil {
					errors = append(errors, GetQuoteResponseValidationError{
						field:  "Quote",
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			}
		} else if v, ok := interface{}(m.GetQuote()).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return GetQuoteResponseValidationError{
					field:  "Quote",
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	case *GetQuoteResponse_NotFound_:
		if v == nil {
			err := GetQuoteResponseValidationError{
				field:  "Result",
				reason: "oneof value cannot be a typed-nil",
			}
			if !all {
				return err
			}
			errors = append(errors, err)
		}

		if all {
			switch v := interface{}(m.GetNotFound()).(type) {
			case interface{ ValidateAll() error }:
				if err := v.ValidateAll(); err != nil {
					errors = append(errors, GetQuoteResponseValidationError{
						field:  "NotFound",
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			case interface{ Validate() error }:
				if err := v.Validate(); err != nil {
					errors = append(errors, GetQuoteResponseValidationError{
						field:  "NotFound",
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			}
		} else if v, ok := interface{}(m.GetNotFound()).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return GetQuoteResponseValidationError{
					field:  "NotFound",
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	default:
		_ = v // ensures v is used
	}

	if len(errors) > 0 {
		return GetQuoteResponseMultiError(errors)
	}

	return nil
}

// GetQuoteResponseMultiError is an error wrapping multiple validation errors
// returned by GetQuoteResponse.ValidateAll() if the designated constraints
// aren't met.
type GetQuoteResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetQuoteResponseMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetQuoteResponseMultiError) AllErrors() []error { return m }

// GetQuoteResponseValidationError is the validation error returned by
// GetQuoteResponse.Validate if the designated constraints aren't met.
type GetQuoteResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetQuoteResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetQuoteResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetQuoteResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetQuoteResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetQuoteResponseValidationError) ErrorName() string { return "GetQuoteResponseValidationError" }

// Error satisfies the builtin error interface
func (e GetQuoteResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetQuoteResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetQuoteResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetQuoteResponseValidationError{}

// Validate checks the field values on ConfirmPayInRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *ConfirmPayInRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ConfirmPayInRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// ConfirmPayInRequestMultiError, or nil if none found.
func (m *ConfirmPayInRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *ConfirmPayInRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for PaymentIntentId

	// no validation rules for PaymentReference

	// no validation rules for PaymentMethod

	if len(errors) > 0 {
		return ConfirmPayInRequestMultiError(errors)
	}

	return nil
}

// ConfirmPayInRequestMultiError is an error wrapping multiple validation
// errors returned by ConfirmPayInRequest.ValidateAll() if the designated
// constraints aren't met.
type ConfirmPayInRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ConfirmPayInRequestMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ConfirmPayInRequestMultiError) AllErrors() []error { return m }

// ConfirmPayInRequestValidationError is the validation error returned by
// ConfirmPayInRequest.Validate if the designated constraints aren't met.
type ConfirmPayInRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ConfirmPayInRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ConfirmPayInRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ConfirmPayInRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ConfirmPayInRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ConfirmPayInRequestValidationError) ErrorName() string {
	return "ConfirmPayInRequestValidationError"
}

// Error satisfies the builtin error interface
func (e ConfirmPayInRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sConfirmPayInRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ConfirmPayInRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ConfirmPayInRequestValidationError{}

// Validate checks the field values on ConfirmPayInResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *ConfirmPayInResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ConfirmPayInResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// ConfirmPayInResponseMultiError, or nil if none found.
func (m *ConfirmPayInResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *ConfirmPayInResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return ConfirmPayInResponseMultiError(errors)
	}

	return nil
}

// ConfirmPayInResponseMultiError is an error wrapping multiple validation
// errors returned by ConfirmPayInResponse.ValidateAll() if the designated
// constraints aren't met.
type ConfirmPayInResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ConfirmPayInResponseMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ConfirmPayInResponseMultiError) AllErrors() []error { return m }

// ConfirmPayInResponseValidationError is the validation error returned by
// ConfirmPayInResponse.Validate if the designated constraints aren't met.
type ConfirmPayInResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ConfirmPayInResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ConfirmPayInResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ConfirmPayInResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ConfirmPayInResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ConfirmPayInResponseValidationError) ErrorName() string {
	return "ConfirmPayInResponseValidationError"
}

// Error satisfies the builtin error interface
func (e ConfirmPayInResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sConfirmPayInResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ConfirmPayInResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ConfirmPayInResponseValidationError{}

// Validate checks the field values on ConfirmPaymentRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *ConfirmPaymentRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ConfirmPaymentRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// ConfirmPaymentRequestMultiError, or nil if none found.
func (m *ConfirmPaymentRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *ConfirmPaymentRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for PaymentIntentId

	// no validation rules for PaymentReference

	// no validation rules for PaymentMethod

	if all {
		switch v := interface{}(m.GetPayOutAmount()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, ConfirmPaymentRequestValidationError{
					field:  "PayOutAmount",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, ConfirmPaymentRequestValidationError{
					field:  "PayOutAmount",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetPayOutAmount()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return ConfirmPaymentRequestValidationError{
				field:  "PayOutAmount",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	if all {
		switch v := interface{}(m.GetReceipt()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, ConfirmPaymentRequestValidationError{
					field:  "Receipt",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, ConfirmPaymentRequestValidationError{
					field:  "Receipt",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetReceipt()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return ConfirmPaymentRequestValidationError{
				field:  "Receipt",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	if len(errors) > 0 {
		return ConfirmPaymentRequestMultiError(errors)
	}

	return nil
}

// ConfirmPaymentRequestMultiError is an error wrapping multiple validation
// errors returned by ConfirmPaymentRequest.ValidateAll() if the designated
// constraints aren't met.
type ConfirmPaymentRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ConfirmPaymentRequestMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ConfirmPaymentRequestMultiError) AllErrors() []error { return m }

// ConfirmPaymentRequestValidationError is the validation error returned by
// ConfirmPaymentRequest.Validate if the designated constraints aren't met.
type ConfirmPaymentRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ConfirmPaymentRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ConfirmPaymentRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ConfirmPaymentRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ConfirmPaymentRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ConfirmPaymentRequestValidationError) ErrorName() string {
	return "ConfirmPaymentRequestValidationError"
}

// Error satisfies the builtin error interface
func (e ConfirmPaymentRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sConfirmPaymentRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ConfirmPaymentRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ConfirmPaymentRequestValidationError{}

// Validate checks the field values on ConfirmPaymentResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *ConfirmPaymentResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ConfirmPaymentResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// ConfirmPaymentResponseMultiError, or nil if none found.
func (m *ConfirmPaymentResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *ConfirmPaymentResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return ConfirmPaymentResponseMultiError(errors)
	}

	return nil
}

// ConfirmPaymentResponseMultiError is an error wrapping multiple validation
// errors returned by ConfirmPaymentResponse.ValidateAll() if the designated
// constraints aren't met.
type ConfirmPaymentResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ConfirmPaymentResponseMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ConfirmPaymentResponseMultiError) AllErrors() []error { return m }

// ConfirmPaymentResponseValidationError is the validation error returned by
// ConfirmPaymentResponse.Validate if the designated constraints aren't met.
type ConfirmPaymentResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ConfirmPaymentResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ConfirmPaymentResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ConfirmPaymentResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ConfirmPaymentResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ConfirmPaymentResponseValidationError) ErrorName() string {
	return "ConfirmPaymentResponseValidationError"
}

// Error satisfies the builtin error interface
func (e ConfirmPaymentResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sConfirmPaymentResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ConfirmPaymentResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ConfirmPaymentResponseValidationError{}

// Validate checks the field values on RejectPaymentIntentRequest with the
// rules defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *RejectPaymentIntentRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on RejectPaymentIntentRequest with the
// rules defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// RejectPaymentIntentRequestMultiError, or nil if none found.
func (m *RejectPaymentIntentRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *RejectPaymentIntentRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for PaymentIntentId

	// no validation rules for PaymentReference

	// no validation rules for Reason

	if len(errors) > 0 {
		return RejectPaymentIntentRequestMultiError(errors)
	}

	return nil
}

// RejectPaymentIntentRequestMultiError is an error wrapping multiple
// validation errors returned by RejectPaymentIntentRequest.ValidateAll() if
// the designated constraints aren't met.
type RejectPaymentIntentRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m RejectPaymentIntentRequestMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m RejectPaymentIntentRequestMultiError) AllErrors() []error { return m }

// RejectPaymentIntentRequestValidationError is the validation error returned
// by RejectPaymentIntentRequest.Validate if the designated constraints aren't met.
type RejectPaymentIntentRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e RejectPaymentIntentRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e RejectPaymentIntentRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e RejectPaymentIntentRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e RejectPaymentIntentRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e RejectPaymentIntentRequestValidationError) ErrorName() string {
	return "RejectPaymentIntentRequestValidationError"
}

// Error satisfies the builtin error interface
func (e RejectPaymentIntentRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sRejectPaymentIntentRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = RejectPaymentIntentRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = RejectPaymentIntentRequestValidationError{}

// Validate checks the field values on RejectPaymentIntentResponse with the
// rules defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *RejectPaymentIntentResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on RejectPaymentIntentResponse with the
// rules defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// RejectPaymentIntentResponseMultiError, or nil if none found.
func (m *RejectPaymentIntentResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *RejectPaymentIntentResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return RejectPaymentIntentResponseMultiError(errors)
	}

	return nil
}

// RejectPaymentIntentResponseMultiError is an error wrapping multiple
// validation errors returned by RejectPaymentIntentResponse.ValidateAll() if
// the designated constraints aren't met.
type RejectPaymentIntentResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m RejectPaymentIntentResponseMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m RejectPaymentIntentResponseMultiError) AllErrors() []error { return m }

// RejectPaymentIntentResponseValidationError is the validation error returned
// by RejectPaymentIntentResponse.Validate if the designated constraints
// aren't met.
type RejectPaymentIntentResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e RejectPaymentIntentResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e RejectPaymentIntentResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e RejectPaymentIntentResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e RejectPaymentIntentResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e RejectPaymentIntentResponseValidationError) ErrorName() string {
	return "RejectPaymentIntentResponseValidationError"
}

// Error satisfies the builtin error interface
func (e RejectPaymentIntentResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sRejectPaymentIntentResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = RejectPaymentIntentResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = RejectPaymentIntentResponseValidationError{}

// Validate checks the field values on
// CreatePaymentIntentResponse_PaymentMethod with the rules defined in the
// proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *CreatePaymentIntentResponse_PaymentMethod) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on
// CreatePaymentIntentResponse_PaymentMethod with the rules defined in the
// proto definition for this message. If any rules are violated, the result is
// a list of violation errors wrapped in
// CreatePaymentIntentResponse_PaymentMethodMultiError, or nil if none found.
func (m *CreatePaymentIntentResponse_PaymentMethod) ValidateAll() error {
	return m.validate(true)
}

func (m *CreatePaymentIntentResponse_PaymentMethod) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for PaymentUrl

	// no validation rules for ProviderId

	// no validation rules for PaymentMethod

	if len(errors) > 0 {
		return CreatePaymentIntentResponse_PaymentMethodMultiError(errors)
	}

	return nil
}

// CreatePaymentIntentResponse_PaymentMethodMultiError is an error wrapping
// multiple validation errors returned by
// CreatePaymentIntentResponse_PaymentMethod.ValidateAll() if the designated
// constraints aren't met.
type CreatePaymentIntentResponse_PaymentMethodMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m CreatePaymentIntentResponse_PaymentMethodMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m CreatePaymentIntentResponse_PaymentMethodMultiError) AllErrors() []error { return m }

// CreatePaymentIntentResponse_PaymentMethodValidationError is the validation
// error returned by CreatePaymentIntentResponse_PaymentMethod.Validate if the
// designated constraints aren't met.
type CreatePaymentIntentResponse_PaymentMethodValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e CreatePaymentIntentResponse_PaymentMethodValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e CreatePaymentIntentResponse_PaymentMethodValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e CreatePaymentIntentResponse_PaymentMethodValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e CreatePaymentIntentResponse_PaymentMethodValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e CreatePaymentIntentResponse_PaymentMethodValidationError) ErrorName() string {
	return "CreatePaymentIntentResponse_PaymentMethodValidationError"
}

// Error satisfies the builtin error interface
func (e CreatePaymentIntentResponse_PaymentMethodValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sCreatePaymentIntentResponse_PaymentMethod.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = CreatePaymentIntentResponse_PaymentMethodValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = CreatePaymentIntentResponse_PaymentMethodValidationError{}

// Validate checks the field values on GetQuoteResponse_Quote with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *GetQuoteResponse_Quote) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetQuoteResponse_Quote with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetQuoteResponse_QuoteMultiError, or nil if none found.
func (m *GetQuoteResponse_Quote) ValidateAll() error {
	return m.validate(true)
}

func (m *GetQuoteResponse_Quote) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if all {
		switch v := interface{}(m.GetRate()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, GetQuoteResponse_QuoteValidationError{
					field:  "Rate",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, GetQuoteResponse_QuoteValidationError{
					field:  "Rate",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetRate()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return GetQuoteResponse_QuoteValidationError{
				field:  "Rate",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	if all {
		switch v := interface{}(m.GetExpiration()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, GetQuoteResponse_QuoteValidationError{
					field:  "Expiration",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, GetQuoteResponse_QuoteValidationError{
					field:  "Expiration",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetExpiration()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return GetQuoteResponse_QuoteValidationError{
				field:  "Expiration",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	if len(errors) > 0 {
		return GetQuoteResponse_QuoteMultiError(errors)
	}

	return nil
}

// GetQuoteResponse_QuoteMultiError is an error wrapping multiple validation
// errors returned by GetQuoteResponse_Quote.ValidateAll() if the designated
// constraints aren't met.
type GetQuoteResponse_QuoteMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetQuoteResponse_QuoteMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetQuoteResponse_QuoteMultiError) AllErrors() []error { return m }

// GetQuoteResponse_QuoteValidationError is the validation error returned by
// GetQuoteResponse_Quote.Validate if the designated constraints aren't met.
type GetQuoteResponse_QuoteValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetQuoteResponse_QuoteValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetQuoteResponse_QuoteValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetQuoteResponse_QuoteValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetQuoteResponse_QuoteValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetQuoteResponse_QuoteValidationError) ErrorName() string {
	return "GetQuoteResponse_QuoteValidationError"
}

// Error satisfies the builtin error interface
func (e GetQuoteResponse_QuoteValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetQuoteResponse_Quote.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetQuoteResponse_QuoteValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetQuoteResponse_QuoteValidationError{}

// Validate checks the field values on GetQuoteResponse_NotFound with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *GetQuoteResponse_NotFound) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetQuoteResponse_NotFound with the
// rules defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetQuoteResponse_NotFoundMultiError, or nil if none found.
func (m *GetQuoteResponse_NotFound) ValidateAll() error {
	return m.validate(true)
}

func (m *GetQuoteResponse_NotFound) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return GetQuoteResponse_NotFoundMultiError(errors)
	}

	return nil
}

// GetQuoteResponse_NotFoundMultiError is an error wrapping multiple validation
// errors returned by GetQuoteResponse_NotFound.ValidateAll() if the
// designated constraints aren't met.
type GetQuoteResponse_NotFoundMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetQuoteResponse_NotFoundMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetQuoteResponse_NotFoundMultiError) AllErrors() []error { return m }

// GetQuoteResponse_NotFoundValidationError is the validation error returned by
// GetQuoteResponse_NotFound.Validate if the designated constraints aren't met.
type GetQuoteResponse_NotFoundValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetQuoteResponse_NotFoundValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetQuoteResponse_NotFoundValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetQuoteResponse_NotFoundValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetQuoteResponse_NotFoundValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetQuoteResponse_NotFoundValidationError) ErrorName() string {
	return "GetQuoteResponse_NotFoundValidationError"
}

// Error satisfies the builtin error interface
func (e GetQuoteResponse_NotFoundValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetQuoteResponse_NotFound.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetQuoteResponse_NotFoundValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetQuoteResponse_NotFoundValidationError{}
